src/fastertransformer/th_op/multi_gpu_gptneox/ParallelGptNeoXOp.h

```diff
@@ -0,0 +1,361 @@
+                                                attention_type,         // attention_type
+                                                nullptr,                // custom_all_reduce_comm
+                                                0                       // enable_custom_all_reduce
+                                                );
+        std::vector<uint32_t> output_seq_len(request_batch_size, total_output_len);
+
+        std::unordered_map<std::string, ft::Tensor> input_tensors = std::unordered_map<std::string, ft::Tensor>{
+            {"input_ids",
+             ft::Tensor{ft::MEMORY_GPU,
+                        ft::TYPE_INT32,
+                        std::vector<size_t>{request_batch_size, max_input_length},
+                        get_ptr<int>(input_ids)}},
+            {"input_lengths",
+             ft::Tensor{
+                 ft::MEMORY_GPU, ft::TYPE_INT32, std::vector<size_t>{request_batch_size}, get_ptr<int>(input_lengths)}},
+            {"output_seq_len",
+             ft::Tensor{
+                 ft::MEMORY_CPU, ft::TYPE_UINT32, std::vector<size_t>{request_batch_size}, output_seq_len.data()}}};
+        if (beam_width > 1 && beam_search_diversity_rate_opt.has_value()) {
+            input_tensors.insert(
+                {"beam_search_diversity_rate",
+                 convert_tensor<float>(beam_search_diversity_rate_opt.value(), ft::MemoryType::MEMORY_CPU)});
+        }
+        if (top_p_opt.has_value()) {
+            input_tensors.insert(
+                {"runtime_top_p", convert_tensor<float>(top_p_opt.value(), ft::MemoryType::MEMORY_CPU)});
+        }
+        if (top_k_opt.has_value()) {
+            input_tensors.insert(
+                {"runtime_top_k", convert_tensor<uint>(top_k_opt.value(), ft::MemoryType::MEMORY_CPU)});
+        }
+        if (temperature_opt.has_value()) {
+            input_tensors.insert(
+                {"temperature", convert_tensor<float>(temperature_opt.value(), ft::MemoryType::MEMORY_CPU)});
+        }
+        if (len_penalty_opt.has_value()) {
+            input_tensors.insert(
+                {"len_penalty", convert_tensor<float>(len_penalty_opt.value(), ft::MemoryType::MEMORY_CPU)});
+        }
+        if (repetition_penalty_opt.has_value()) {
+            input_tensors.insert({"repetition_penalty",
+                                  convert_tensor<float>(repetition_penalty_opt.value(), ft::MemoryType::MEMORY_CPU)});
+        }
+        if (random_seed_opt.has_value()) {
+            input_tensors.insert(
+                {"random_seed",
+                 convert_tensor<unsigned long long int>(random_seed_opt.value(), ft::MemoryType::MEMORY_CPU)});
+        }
+
+        bool return_context_cum_log_probs = false;
+        if (return_cum_log_probs == 2) {
+            return_context_cum_log_probs = true;
+            input_tensors.insert(
+                {"is_return_context_cum_log_probs",
```

Comment:
Please check the GptNeoXTritonModelInstance to make sure what inputs are supported. For example, `is_return_context_cum_log_probs` is only supported in GPT and is not supported in Gpt Neox.